### PR TITLE
Creates onboard_scan_target method addressing issue #27

### PR DESCRIPTION
**Pull request by leonardoviveiros**
## Goal
This PR adds the onboard_scan_target method to the ZanshinSDK that onboards new AWS Accounts via SDK.

## Tests
This was tested manually using Zanshin Dev and Prod environment. The tenchi-it AWS account was onboarded.
There're also unit tests for the following situation:
- Onboard of ScanTargets other then AWS expects an Exception.
- Onboard using invalid AWS Credentials will fail.
- Onboard without `boto3` dependency will fail. This dependency is not a requirement for other SDK methods, just for this one.
- Onboard succeeded and a new Scan Target is created and the cloudformation deployed to AWS.

## Dependencies
For the tests to run, we need to install `moto[all]` and `boto3`. These dependencies are not necessary to run the sdk normally. Only `boto3` is required for only the new method `onboard_scan_target`

Issue number from the [MegaBoard](https://github.com/orgs/tenchi-security/projects/14/views/10) #120 from the Megaboard

New stuff:
-  Add method onboard_scan_target with tests. 
- Add options to install zanshinsdk[with_boto3].
- Add moto as framework to mock boto3.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,8 +46,26 @@ me = client.get_me()    # calls /me API endpoint
...
+For more examples, checkout the [docs](zanshinsdk/docs/README.md).
+
 All operations call `raise_for_status` on the httpx [Response object](https://www.python-httpx.org/api/#response) internally, so any 4xx or 5xx will raise [exceptions](https://www.python-httpx.org/exceptions/).
 
+## Installing
+
+To install the SDK, you can use `pip`. You have two options to install ZanshinSDK:
+- *Essentials*
+  
+Using `pip install zanshinsdk` will install the SDK with all features exception ability to perform onboarding of new Scan Targets. For this, you'll need to install boto3.
```
Comment (AlisonVilela) [inline README.md:58]: It is possible to create using the create_organization_scan_target function, I believe that using boto3 would be the automated onboarding that includes running the cloudformation

File: zanshinsdk/test_client.py [hunk 2]
```diff
@@ -2068,3 +2081,120 @@ def test_validate_uuid_invalid(self):
...
+        self.mock_aws_credentials()
+        
+        # Mock request to create new Scan Target
+        mock_is_file.return_value = True
+        _data = f"[default]\napi_key=api_key"
+
+        with patch("__main__.__builtins__.open", mock_open(read_data=_data)):
+            request.return_value = Mock(status_code=200, json=lambda: {
+                                        "id": created_scan_target_id})
+            self.sdk = zanshinsdk.Client()
```
Comment (AlisonVilela) [inline zanshinsdk/test_client.py:2163]: It is preferable to use a local variable instead of overwriting the instance created in setUp

Comment (leonardoviveiros) [reply zanshinsdk/test_client.py]: Fixed! Thanks.

File: README.md [hunk 3]
```diff
@@ -46,8 +46,26 @@ me = client.get_me()    # calls /me API endpoint
...
+
+- *With Boto3*
+
+With `pip install zanshinsdk[with_boto3]` you'll automatically install [boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html) along with ZanshinSDK. This will enable you to perform Onboard of new Scan Targets via SDK.
+
+## Testing
+
+To run all tests call `python -m unittest discover -v` on the project root directory. Make sure there's a `[default]` profile configured, else some tests will fail.
```
[the remark this answers is not in the document]
Comment (leonardoviveiros) [reply README.md]: Fixed! Thanks.

**Pull request merged**